api/src/OpenTelemetry/Internal/Common/Types.hs

```diff
@@ -59,3 +71,85 @@ instance Hashable InstrumentationLibrary
 instance IsString InstrumentationLibrary where
   fromString :: String -> InstrumentationLibrary
   fromString str = InstrumentationLibrary (fromString str) "" "" emptyAttributes
+
+
+{- | An attribute represents user-provided metadata about a span, link, or event.
+
+ 'Any' values are used in place of 'Standard Attributes' in logs because third-party
+ logs may not conform to the 'Standard Attribute' format.
+
+ Telemetry tools may use this data to support high-cardinality querying, visualization
+ in waterfall diagrams, trace sampling decisions, and more.
+-}
+data AnyValue
+  = TextValue Text
+  | BoolValue Bool
+  | DoubleValue Double
+  | IntValue Int64
+  | ByteStringValue ByteString
+  | ArrayValue [AnyValue]
+  | HashMapValue (H.HashMap Text AnyValue)
+  | NullValue
```

Comment:
Added NullValue [in accordance with the spec](https://opentelemetry.io/docs/specs/otel/logs/data-model/#type-any).